crates/ruff/src/autofix/mod.rs

```diff
@@ -96,51 +96,50 @@ pub(crate) fn apply_fix(fix: &Fix, locator: &Locator) -> String {
 mod tests {
     use rustpython_parser::ast::Location;
 
+    use ruff_diagnostics::Diagnostic;
+    use ruff_diagnostics::Fix;
     use ruff_python_ast::source_code::Locator;
 
     use crate::autofix::{apply_fix, apply_fixes};
-    use crate::fix::Fix;
-    use crate::registry::Diagnostic;
     use crate::rules::pycodestyle::rules::NoNewLineAtEndOfFile;
 
+    fn as_diagnostic(fix: Fix) -> Diagnostic {
```

Comment:
Nit: Rename to `into_diagnostic`

![image](https://user-images.githubusercontent.com/1203881/223984303-f8c55425-a14a-435b-94e2-3e321d4a1515.png)

[Source](https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv)


Or you can change it to the following `create_diagnostics` function that avoids the need for the vector:

```rust
fn create_diagnostics(fixes: impl IntoIterator<Item=Fix>) -> Vec<Diagnostic> {
  fixes.into_iter().map(|fix| Diagnostic {
        // The choice of rule here is arbitrary.
        kind: NoNewLineAtEndOfFile.into(),
        location: fix.location,
        end_location: fix.end_location,
        fix: Some(fix),
        parent: None,
    }).collect()
}
```

Usage: 

```rust
let empty_diagnostics = create_diagnostics([]);
let non_empty = create_diagnostics([Fix {
    content: "Bar".to_string(),
    location: Location::new(1, 8),
    end_location: Location::new(1, 14),
}])
```

Reply:
Awesome, thank you!